tests/ssl-opt.sh

```diff
@@ -672,6 +672,9 @@ run_test() {
     # as it provides timing info that's useful to debug failures
     if [ -z "$PXY_CMD" ] && [ "$DTLS" -eq 1 ]; then
         PXY_CMD="$P_PXY"
+        if echo "$SRV_CMD" | grep 'server_addr=::1' >/dev/null; then
```

Comment:
Our heavy use of `grep` has a non-negligible performance penalty. For future reference, as long as simple wildcard patterns are enough (as opposed to full-blown regexps), you can use a shell built-in construct, which is faster.

```
case "$SRV_CMD" in
  *'server_addr=::1'*) PXY_CMD="$PXY_CMD server_addr=::1 listen_addr=::1";;
esac
```

Reply:
I tried that first, but apparently I messed it up because it didn't work, so I lazily fell back to grep. Looking at your code I think my mistake was I forgot the quotes in `*'server_addr=::1'*`. Now I know for next time.

Reply:
In this case the quotes aren't necessary since none of the characters in `server_addr=::1` have a special meaning.

Reply:
Hmm, I just checked the spec, and none of the characters here seem to require quoting. I also just tried it without the quotes and it works, so I must have made another mistake. Anyway, I just amended the last commit, I hope you don't mind. Compared to your suggestion, I've also added spaces to only match on a whole word.

Reply:
(Wrote my comment before seeing yours, good that we came to the same conclusion.)